src/snapshot.rs

```diff
@@ -79,17 +80,20 @@ pub struct SnapshotCreator([usize; 1]);
 impl SnapshotCreator {
   /// Create and enter an isolate, and set it up for serialization.
   /// The isolate is created from scratch.
-  pub fn new(external_references: Option<&'static ExternalReferences>) -> Self {
+  pub fn new(
+    external_references: Option<&'static ExternalReferences>,
+    startup_data: Option<StartupData>,
```

Comment:
Downside of this setup is that the caller must own the snapshot (it can't be `&'static StartupData` or `Rc<StartupData>`), and therefore sharing the snapshot blob with another Isolate or SnapshotCreator. It is the direct opposite of `external_references` which takes *only* a reference and not an owned value.